File: practice/algorithms/implementation/divisible_sum_pairs/DivisibleSumPairs.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const line1 = lines[0];
    const divisor = parseInt(line1.split(' ')[1]);
    const array = lines[1].split(' ').map(item => parseInt(item));
    let count = 0;
    for(let i = 0; i < array.length - 1; i++) {
        for(let j = i + 1; j < array.length; j++) {
            if((array[i] + array[j]) % divisor === 0) {
                count++;
            }
        }
    }
    console.log(count);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/divisible_sum_pairs/divisible_sum_pairs.rb
(n, divisor) = gets.split(" ").map {|i| i.to_i}
array = gets.split(" ").map {|i| i.to_i}
count = 0
for a in 0..(n-2) do
  for j in (a+1)..(n-1) do
    count += 1 if ((array[a] + array[j]) % divisor == 0)
  end
end

puts count